tests/benchmark/benchmark_bettertransformer.py

```diff
@@ -90,41 +100,46 @@ def get_batch(batch_size, avg_seqlen, max_sequence_length, seqlen_stdev, vocab_s
     return tokens, lengths, mask
 
 
-def timing_cuda(model, num_batches, input_ids, masks):
+def timing_cuda(model, num_batches, input_ids, masks, is_decoder, generation_config=None):
     start_event = torch.cuda.Event(enable_timing=True)
     end_event = torch.cuda.Event(enable_timing=True)
     start_event.record()
     for _ in range(num_batches):
-        _ = model(input_ids, masks)
+        if is_decoder:
+            _ = model.generate(input_ids, generation_config=generation_config)
+        else:
+            _ = model(input_ids, masks)
     end_event.record()
     torch.cuda.synchronize()
     return (start_event.elapsed_time(end_event) * 1.0e-3) / num_batches
 
 
-def benchmark(model_name, num_batches, batch_size, avg_seqlen, max_seqlen, seqlen_stdev, is_cuda, is_half, use_mask):
-    print("Loading model {}".format(model_name))
-    hf_model = AutoModel.from_pretrained(model_name, torch_dtype=torch.float16 if is_half else None).eval()
-    if is_cuda:
-        hf_model = hf_model.to(0)
-    bt_model = BetterTransformer.transform(hf_model, keep_original_model=True)
-
-    input_ids, _, masks = get_batch(batch_size, avg_seqlen, max_seqlen, seqlen_stdev)
-
-    if is_cuda:
-        input_ids = input_ids.to(0)
-        masks = masks.to(0)
-
-    if not use_mask:
-        masks = None
-
+def benchmark(hf_model, bt_model, num_batches, is_decoder, max_token):
     # Warmup
-    _ = hf_model(input_ids, masks)
-    torch.cuda.synchronize()
-    _ = bt_model(input_ids, masks)
-    torch.cuda.synchronize()
-
-    total_hf_time = timing_cuda(hf_model, num_batches, input_ids, masks)
-    total_bt_time = timing_cuda(bt_model, num_batches, input_ids, masks)
+    if is_decoder:
+        gen_config = GenerationConfig(
+            do_greedy=True,
+            max_new_tokens=max_token,
+            use_cache=False,
```

Comment:
Why `use_cache = False`? I'd rather benchmark with the cache enabled.

Also, can we rather set `min_length=length, max_length=length`? So that we can generate a specific length.

Reply:
I think it's because `max_length` is deprecated, yes we can add `min_length` I think let me try